sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/CheckpointPosition.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Consumer
```

Comment:
We should probably move this to the `Processor` namespace with args types and other processor-related stuff, as the checkpoint concept doesn't map to a standard consumer.